src/main/java/cstwooneohthree/glados/utils/Ui.java

```diff
@@ -0,0 +1,199 @@
+                         + LOGO + "What task would you like me to perform today?" + "\n";
+
+        if (uiType == UiType.COMMAND_LINE_INTERFACE) {
+            System.out.println(HORIZONTAL_LINE + message + HORIZONTAL_LINE);
+            return null;
+        } else {
+            return "Hello, welcome to the Aperture Science computer-aided enrichment center!"
+                    + "My name is GLaDOS. What task would you like me to perform today?";
+        }
+    }
+
+    /**
+     * Prints or returns user input in echo command.
+     *
+     * @param input Input to be echoed.
+     */
+    public String echo(String input) {
+        String message = "\nGLaDOS: " + input.trim() + "\n";
+
+        if (uiType == UiType.COMMAND_LINE_INTERFACE) {
+            System.out.println(HORIZONTAL_LINE + message + HORIZONTAL_LINE);
+            return null;
+        } else {
+            return input.trim();
+        }
+    }
+
+    /**
+     * Prints or returns result after task has been added.
+     *
+     * @param task Task description to be added.
+     * @param index Number of tasks left.
+     */
+    public String add(String task, String index) {
+        String message = "\nGLaDOS: I have added the following task to the list...\n" 
+                        + task + "\n" 
+                        + "Now you have " + index + (Integer.parseInt(index) == 1 ? " task." : " tasks.")
+                        + "\n";
+
+        if (uiType == UiType.COMMAND_LINE_INTERFACE) {
+            System.out.println(HORIZONTAL_LINE + message + HORIZONTAL_LINE);
+            return null;
+        } else {
+            return "I have added the following task to the list: " 
+                   + task + " Now you have " + index + (Integer.parseInt(index) == 1 ? " task." : " tasks.");
+        }
+    }
+
+    /**
+     * Prints or returns result after task has been deleted.
+     *
+     * @param task Task description to be deleted.
+     * @param index Number of tasks left.
+     */
+    public String delete(String task, String index) {
+        String message = "\nGLaDOS: I have removed the following task from the list...\n"
+                        + task + "\n"
+                        + "Now you have " + index + (Integer.parseInt(index) == 1 ? " task." : " tasks.")
+                        + "\n";
+
+        if (uiType == UiType.COMMAND_LINE_INTERFACE) {
+            System.out.println(HORIZONTAL_LINE + message + HORIZONTAL_LINE);
+            return null;
+        } else {
+            return "I have removed the following task from the list: " 
+                   + task + " Now you have " + index + (Integer.parseInt(index) == 1 ? " task." : " tasks.");
+        }
+    }
+
+    /**
+     * Prints or returns all elements inside the tasklist.
+     *
+     * @param taskList Task list to be printed.
+     */
+    public String list(ArrayList<Task> taskList, boolean shouldFindMatching) {
```

Comment:
I thought shouldFindmatching might be slightly ambiguous? Not quite clear what it does. Maybe explain in the docstring or a comment